homeassistant/components/google_travel_time/sensor.py

```diff
@@ -109,77 +152,127 @@ async def async_added_to_hass(self) -> None:
             await self.first_update()
 
     @property
-    def native_value(self):
+    def native_value(self) -> float | None:
         """Return the state of the sensor."""
-        if self._matrix is None:
+        if self._route is None:
             return None
 
-        _data = self._matrix["rows"][0]["elements"][0]
-        if "duration_in_traffic" in _data:
-            return round(_data["duration_in_traffic"]["value"] / 60)
-        if "duration" in _data:
-            return round(_data["duration"]["value"] / 60)
-        return None
+        return round(self._route.duration.seconds / 60)
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, Any] | None:
         """Return the state attributes."""
-        if self._matrix is None:
+        if self._route is None:
             return None
 
-        res = self._matrix.copy()
-        options = self._config_entry.options.copy()
-        res.update(options)
-        del res["rows"]
-        _data = self._matrix["rows"][0]["elements"][0]
-        if "duration_in_traffic" in _data:
-            res["duration_in_traffic"] = _data["duration_in_traffic"]["text"]
-        if "duration" in _data:
-            res["duration"] = _data["duration"]["text"]
-        if "distance" in _data:
-            res["distance"] = _data["distance"]["text"]
-        res["origin"] = self._resolved_origin
-        res["destination"] = self._resolved_destination
-        return res
-
-    async def first_update(self, _=None):
+        result = self._config_entry.options.copy()
+        result["duration_in_traffic"] = self._route.localized_values.duration.text
+        result["duration"] = self._route.localized_values.static_duration.text
+        result["distance"] = self._route.localized_values.distance.text
+
+        result["origin"] = self._resolved_origin
+        result["destination"] = self._resolved_destination
+        return result
```

Comment:
it would be nice to create separate entities for this in the future

Reply:
Fully agree. Not the scope of this PR though

Reply:
Yep, agreed, but I do want to have it mentioned :)